src/common/io-config/src/s3.rs

```diff
@@ -44,6 +56,10 @@ impl S3Config {
         if let Some(access_key) = &self.access_key {
             res.push(format!("Access key = {}", access_key));
         }
+        #[cfg(feature = "python")]
```

Comment:
this is usually hard to reason around! We should just have the feature gate inside `credentials_provider` for any specific python types you have.